redis/src/connection.rs

```diff
@@ -1014,6 +1024,31 @@ pub(crate) fn connection_setup_pipeline(
     )
 }
 
+#[cfg(feature = "cache-aio")]
+fn client_caching_setup(
+    connection_info: &RedisConnectionInfo,
+    cache_config: crate::caching::CacheConfig,
+    pipeline: &mut Pipeline,
+) {
+    if connection_info.protocol == ProtocolVersion::RESP2 {
+        return;
+    }
+    match cache_config.mode {
+        crate::caching::CacheMode::None => {}
+        crate::caching::CacheMode::All => {
+            pipeline.cmd("CLIENT").arg("TRACKING").arg("ON").ignore();
```

Comment:
I prefer that caching won't be silently ignored. If the user turns on caching with a client level <6, they should receive an informative error message.

Reply:
Now it will raise an error